mne/io/meas_info.py

```diff
@@ -815,37 +815,57 @@ def ch_names(self):
         return self['ch_names']
 
     def _repr_html_(self, caption=None):
+        """Summarize info for HTML representation."""
         if isinstance(caption, str):
             html = f'<h4>{caption}</h4>'
         else:
             html = ''
-        n_eeg = len(pick_types(self, meg=False, eeg=True))
-        n_grad = len(pick_types(self, meg='grad'))
-        n_mag = len(pick_types(self, meg='mag'))
-        n_fnirs = len(pick_types(self, meg=False, eeg=False, fnirs=True))
-        pick_eog = pick_types(self, meg=False, eog=True)
-        if len(pick_eog) > 0:
-            eog = ', '.join(np.array(self['ch_names'])[pick_eog])
-        else:
+
+        # good channels
+        channels = {}
+        ch_types = [channel_type(self, idx) for idx in range(len(self['chs']))]
+        ch_counts = Counter(ch_types)
+        for ch_type, count in ch_counts.items():
+            if ch_type == 'meg':
+                channels['mag'] = len(pick_types(self, meg='mag'))
+                channels['grad'] = len(pick_types(self, meg='grad'))
+            elif ch_type == 'eog':
+                pick_eog = pick_types(self, meg=False, eog=True)
```

Comment:
`meg=False` has been the default long enough I don't think we need it anymore. It was just there for the transition...
```suggestion
                pick_eog = pick_types(self, eog=True)
```